croncheck.php

```diff
@@ -121,8 +121,45 @@
     header('Expires: Tue, 04 Sep 2012 05:32:29 GMT');
 }
 
+if (isset($CFG->adminsetuppending)) {
+    send_critical("Admin setup pending, please set up admin user account");
+}
+
 if (moodle_needs_upgrading()) {
-    send_critical("Moodle upgrade pending, cron execution suspended");
+    // Check if an upgrade is running.
+    if (!isset($CFG->upgraderunning) || $CFG->upgraderunning < time()) {
+        $upgraderunning = get_config(null, 'upgraderunning');
+    } else {
+        $upgraderunning = $CFG->upgraderunning;
```

Comment:
It is for the rare case if $CFG->upgraderunning is set in the config.php, as a result it will not be set in set_config() because "upgraderunning" will be a key in $CFG->config_php_settings:
https://github.com/moodle/moodle/blob/MOODLE_35_STABLE/lib/moodlelib.php#L1392
Then we need to get config from DB
It would be a rare case so I remove the code to make ti simple.